File: gfg/most-frequent-word-in-array.cpp
/*
https://practice.geeksforgeeks.org/problems/most-frequent-word-in-an-array-of-strings/0
 */

#include <bits/stdc++.h>
using namespace std;
#define psi pair<string, int>

bool comp(psi a, psi b)
{
	if(a.second > b.second)
		return true;
	else if(a.second == b.second)
		return a.first < b.first;
	else
		return false;

}
int main()
{
	ios_base::sync_with_stdio(false);
	int t,n;
	for(cin>>t;t--;)
	{
		map<string, int> hash;
		string str;
		cin>>n;
		for(int i=0;i<n;i++)
		{
			cin>>str;
			if (hash.find(str) == hash.end())
				hash[str] = 0;
			hash[str]++;
		}
		vector<psi> hc(hash.begin(), hash.end());
		sort(hc.begin(), hc.end(), comp);
		cout<<hc[0].first<<endl;
	}
}

File: old/gfg/most-frequent-word-in-array.py
# https://practice.geeksforgeeks.org/problems/most-frequent-word-in-an-array-of-strings/0

from collections import Counter

t = int(input())
while t > 0:
    n = int(input())
    words = input().strip().split()
    words = list(Counter(words).items())
    words.sort(key=lambda x: x[1], reverse=True)
    max_freq = words[0][1]
    words = [each for each in words if each[1] == max_freq]
    words.sort(key=lambda x: x[0])
    print(words[0][0])
    t -= 1
